internal/dotc1z/sync_runs.go

```diff
@@ -351,3 +354,93 @@ func (c *C1File) EndSync(ctx context.Context) error {
 
 	return nil
 }
+
+func (c *C1File) Cleanup(ctx context.Context) error {
+	l := ctxzap.Extract(ctx)
+
+	if skipCleanup, _ := strconv.ParseBool(os.Getenv("BATON_SKIP_CLEANUP")); skipCleanup {
+		return nil
+	}
+
+	err := c.validateDb(ctx)
+	if err != nil {
+		return err
+	}
+
+	if c.currentSyncID != "" {
+		return nil
+	}
+
+	var ret []*syncRun
+
+	pageToken := ""
+	for {
+		runs, nextPageToken, err := c.ListSyncRuns(ctx, pageToken, 100)
+		if err != nil {
+			return err
+		}
+
+		for _, sr := range runs {
+			if sr.EndedAt == nil {
+				continue
+			}
+			ret = append(ret, sr)
+		}
+
+		if nextPageToken == "" {
+			break
+		}
+		pageToken = nextPageToken
+	}
+
+	syncLimit := 2
+	if customSyncLimit, err := strconv.ParseInt(os.Getenv("BATON_KEEP_SYNC_COUNT"), 10, 64); err == nil && customSyncLimit > 0 {
+		syncLimit = int(customSyncLimit)
+	}
+
+	if len(ret) <= syncLimit {
+		return nil
+	}
+
+	l.Info("Cleaning up old sync data...")
+	for i := 0; i < len(ret)-syncLimit; i++ {
```

Comment:
small nit: this and the above logic might be a little cleaner with something like:
```
for _, run := range ret[syncLimit:] { ...
```